### ISSUE TITLE
Inhomogeneous shape error when running segmentation

### ISSUE THREAD

**Issue by MatiNem** [Vizgen/vizgen-postprocessing#46]
Hello, 

I seem to be encountering an inhomogeneous shape error when I run vpt run-segmentation on specific files. I have run vpt successfully before but a few files encounter this error:

`/usr/local/lib/python3.10/dist-packages/geopandas/geodataframe.py:1459: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  result = super().__getitem__(key)
2025-10-14 04:05:35,104 - ./task-31 - INFO - remove edge polys
2025-10-14 04:05:38,630 - ./task-31 - INFO - fuse_task_polygons
2025-10-14 04:05:40,320 - ./task-31 - INFO - Found 3533 overlaps
2025-10-14 04:05:54,871 - ./task-94 - INFO - get_polygons_from_mask: z=6, labels:899
2025-10-14 04:05:59,079 - ./task-31 - INFO - After union of large overlaps, found 1531 overlaps
2025-10-14 04:06:11,954 - ./task-31 - INFO - After both resolution steps, found 0 uncaught overlaps
2025-10-14 04:06:13,848 - ./task-31 - INFO - Run segmentation on tile 31 finished
2025-10-14 04:06:13,879 - ./task-298 - INFO - Run segmentation on tile 298 started
2025-10-14 04:06:13,952 - ./task-298 - INFO - Tile 298 [82728, 60612, 5096, 5096]
2025-10-14 04:06:27,494 - ./task-298 - INFO - generate_polygons_from_mask
2025-10-14 04:06:27,669 - ./task-298 - INFO - get_polygons_from_mask: z=0, labels:1
2025-10-14 04:06:27,838 - ./task-298 - INFO - get_polygons_from_mask: z=1, labels:1
2025-10-14 04:06:28,010 - ./task-298 - INFO - get_polygons_from_mask: z=2, labels:1
2025-10-14 04:06:28,179 - ./task-298 - INFO - get_polygons_from_mask: z=3, labels:1
2025-10-14 04:06:28,356 - ./task-298 - INFO - get_polygons_from_mask: z=4, labels:1
2025-10-14 04:06:28,511 - ./task-298 - INFO - get_polygons_from_mask: z=5, labels:1
2025-10-14 04:06:28,684 - ./task-298 - INFO - get_polygons_from_mask: z=6, labels:1
2025-10-14 04:06:28,694 - ./task-298 - INFO - raw segmentation result contains 0 rows
2025-10-14 04:06:28,694 - ./task-298 - INFO - fuze across z
2025-10-14 04:06:28,695 - ./task-298 - INFO - remove edge polys
2025-10-14 04:06:34,421 - ./task-288 - INFO - generate_polygons_from_mask
2025-10-14 04:06:34,913 - ./task-288 - INFO - get_polygons_from_mask: z=0, labels:2174
2025-10-14 04:06:38,108 - ./task-94 - INFO - raw segmentation result contains 6740 rows
2025-10-14 04:06:38,109 - ./task-94 - INFO - fuze across z
2025-10-14 04:06:38,740 - ./task-34 - INFO - get_polygons_from_mask: z=2, labels:1397
2025-10-14 04:06:39,430 - ./task-298 - INFO - generate_polygons_from_mask
2025-10-14 04:06:39,587 - ./task-298 - INFO - get_polygons_from_mask: z=0, labels:1
2025-10-14 04:06:39,747 - ./task-298 - INFO - get_polygons_from_mask: z=1, labels:1
2025-10-14 04:06:39,912 - ./task-298 - INFO - get_polygons_from_mask: z=2, labels:1
2025-10-14 04:06:40,073 - ./task-298 - INFO - get_polygons_from_mask: z=3, labels:1
2025-10-14 04:06:40,234 - ./task-298 - INFO - get_polygons_from_mask: z=4, labels:1
2025-10-14 04:06:40,405 - ./task-298 - INFO - get_polygons_from_mask: z=5, labels:1
2025-10-14 04:06:40,576 - ./task-298 - INFO - get_polygons_from_mask: z=6, labels:1
2025-10-14 04:06:40,581 - ./task-298 - INFO - raw segmentation result contains 0 rows
2025-10-14 04:06:40,582 - ./task-298 - INFO - fuze across z
2025-10-14 04:06:40,582 - ./task-298 - INFO - remove edge polys
2025-10-14 04:06:40,583 - ./task-298 - INFO - fuse_task_polygons
2025-10-14 04:06:40,584 - ./task-298 - INFO - Found 0 overlaps
2025-10-14 04:06:40,584 - ./task-298 - INFO - After union of large overlaps, found 0 overlaps
2025-10-14 04:06:40,584 - ./task-298 - INFO - After both resolution steps, found 0 uncaught overlaps
2025-10-14 04:06:40,619 - ./task-298 - INFO - Run segmentation on tile 298 finished
2025-10-14 04:06:40,620 - ./task-299 - INFO - Run segmentation on tile 299 started
2025-10-14 04:06:40,658 - ./task-299 - INFO - Tile 299 [82886, 60612, 5096, 5096]
/usr/local/lib/python3.10/dist-packages/geopandas/geodataframe.py:1459: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  result = super().__getitem__(key)
2025-10-14 04:06:46,207 - ./task-94 - INFO - remove edge polys
2025-10-14 04:06:49,788 - ./task-94 - INFO - fuse_task_polygons
2025-10-14 04:06:51,590 - ./task-94 - INFO - Found 3204 overlaps
2025-10-14 04:06:53,629 - ./task-299 - INFO - generate_polygons_from_mask
2025-10-14 04:06:53,805 - ./task-299 - INFO - get_polygons_from_mask: z=0, labels:1
2025-10-14 04:06:53,992 - ./task-299 - INFO - get_polygons_from_mask: z=1, labels:1
2025-10-14 04:06:54,168 - ./task-299 - INFO - get_polygons_from_mask: z=2, labels:1
2025-10-14 04:06:54,342 - ./task-299 - INFO - get_polygons_from_mask: z=3, labels:1
2025-10-14 04:06:54,507 - ./task-299 - INFO - get_polygons_from_mask: z=4, labels:1
2025-10-14 04:06:54,667 - ./task-299 - INFO - get_polygons_from_mask: z=5, labels:1
2025-10-14 04:06:54,826 - ./task-299 - INFO - get_polygons_from_mask: z=6, labels:1
2025-10-14 04:06:54,835 - ./task-299 - INFO - raw segmentation result contains 0 rows
2025-10-14 04:06:54,835 - ./task-299 - INFO - fuze across z
2025-10-14 04:06:54,836 - ./task-299 - INFO - remove edge polys
2025-10-14 04:07:05,652 - ./task-299 - INFO - generate_polygons_from_mask
2025-10-14 04:07:05,808 - ./task-299 - INFO - get_polygons_from_mask: z=0, labels:1
2025-10-14 04:07:05,959 - ./task-299 - INFO - get_polygons_from_mask: z=1, labels:1
2025-10-14 04:07:06,103 - ./task-299 - INFO - get_polygons_from_mask: z=2, labels:1
2025-10-14 04:07:06,252 - ./task-299 - INFO - get_polygons_from_mask: z=3, labels:1
2025-10-14 04:07:06,406 - ./task-299 - INFO - get_polygons_from_mask: z=4, labels:1
2025-10-14 04:07:06,548 - ./task-299 - INFO - get_polygons_from_mask: z=5, labels:1
2025-10-14 04:07:06,697 - ./task-299 - INFO - get_polygons_from_mask: z=6, labels:1
2025-10-14 04:07:06,702 - ./task-299 - INFO - raw segmentation result contains 0 rows
2025-10-14 04:07:06,702 - ./task-299 - INFO - fuze across z
2025-10-14 04:07:06,703 - ./task-299 - INFO - remove edge polys
2025-10-14 04:07:06,704 - ./task-299 - INFO - fuse_task_polygons
2025-10-14 04:07:06,704 - ./task-299 - INFO - Found 0 overlaps
2025-10-14 04:07:06,705 - ./task-299 - INFO - After union of large overlaps, found 0 overlaps
2025-10-14 04:07:06,705 - ./task-299 - INFO - After both resolution steps, found 0 uncaught overlaps
2025-10-14 04:07:06,774 - ./task-299 - INFO - Run segmentation on tile 299 finished
2025-10-14 04:07:06,811 - ./task-262 - INFO - Run segmentation on tile 262 started
2025-10-14 04:07:06,831 - ./task-262 - INFO - Tile 262 [9192, 59748, 5096, 5096]
2025-10-14 04:07:10,511 - ./task-94 - INFO - After union of large overlaps, found 1343 overlaps
2025-10-14 04:07:22,266 - ./task-94 - INFO - After both resolution steps, found 0 uncaught overlaps
2025-10-14 04:07:24,155 - ./task-94 - INFO - Run segmentation on tile 94 finished
2025-10-14 04:07:24,173 - ./task-95 - INFO - Run segmentation on tile 95 started
2025-10-14 04:07:24,182 - ./task-95 - INFO - Tile 95 [68940, 18384, 5096, 5096]
2025-10-14 04:07:46,178 - ./task-34 - INFO - get_polygons_from_mask: z=3, labels:977
2025-10-14 04:08:14,945 - ./task-288 - INFO - get_polygons_from_mask: z=1, labels:749
2025-10-14 04:08:35,745 - ./task-34 - INFO - get_polygons_from_mask: z=4, labels:935
2025-10-14 04:08:50,627 - ./task-288 - INFO - get_polygons_from_mask: z=2, labels:415
2025-10-14 04:09:11,967 - ./task-288 - INFO - get_polygons_from_mask: z=3, labels:273
2025-10-14 04:09:22,122 - ./task-34 - INFO - get_polygons_from_mask: z=5, labels:907
2025-10-14 04:09:24,629 - ./task-288 - INFO - get_polygons_from_mask: z=4, labels:246
2025-10-14 04:09:33,935 - ./task-37 - ERROR - exception of type <class 'ValueError'> thrown: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (7, 2) + inhomogeneous part.
2025-10-14 04:09:33,966 - distributed.worker - ERROR - Compute Failed
Key:       ('lambda-8a002f3cb22e8f88ec84080597141e06', 18)
State:     executing
Function:  execute_task
args:      ((<function reify at 0x7f3e605c3520>, (<function map_chunk at 0x7f3e605c3910>, <function Context.parallel_run.<locals>.<lambda> at 0x7f3de92ee680>, [[{'task': Task(proc=<function run at 0x7f3e606b9750>, args=Namespace(input_segmentation_parameters='/storage2/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/new_output/region_0//segmentation_specification.json', tile_index=36, overwrite=True)), 'cnt_args': {'name': './task-36', 'dask_args': {'workers': 8}, 'fs_args': {'aws_profile_name': None, 'aws_access_key': None, 'aws_secret_key': None, 'gcs_service_account_key': None}, 'log_args': {'fname': None, 'lvl': 2, 'verbose': True}, 'prof_args': {'profile_file': None}}}, {'task': Task(proc=<function run at 0x7f3e606b9750>, args=Namespace(input_segmentation_parameters='/storage2/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/new_output/region_0//segmentation_specification.json', tile_index=37, overwrite=True)), 'cnt_args': {'name': './task
kwargs:    {}
Exception: "ValueError('setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (7, 2) + inhomogeneous part.')"
Traceback: '  File "/usr/local/lib/python3.10/dist-packages/dask/bag/core.py", line 1868, in reify\n    seq = list(seq)\n  File "/usr/local/lib/python3.10/dist-packages/dask/bag/core.py", line 2056, in __next__\n    return self.f(*vals)\n  File "/usr/local/lib/python3.10/dist-packages/vpt/app/context.py", line 166, in <lambda>\n    mp_bag = mp_bag.map(lambda b: _context_wrapper(**b))\n  File "/usr/local/lib/python3.10/dist-packages/vpt/app/context.py", line 163, in _context_wrapper\n    return task.proc(task.args)\n  File "/usr/local/lib/python3.10/dist-packages/vpt/run_segmentation_on_tile/__init__.py", line 11, in run\n    run_segmentation_on_tile(args)\n  File "/usr/local/lib/python3.10/dist-packages/vpt/run_segmentation_on_tile/main.py", line 113, in run_segmentation_on_tile\n    result = segmentation_on_tile(seg_spec, args.tile_index)\n  File "/usr/local/lib/python3.10/dist-packages/vpt/run_segmentation_on_tile/main.py", line 87, in segmentation_on_tile\n    tasks_result = get_tile_segmentation(seg_spec, window_info)\n  File "/usr/local/lib/python3.10/dist-packages/vpt/run_segmentation_on_tile/main.py", line 26, in get_tile_segmentation\n    seg_result = runner(\n  File "/usr/local/lib/python3.10/dist-packages/vpt_plugin_cellpose/segment.py", line 26, in run_segmentation\n    masks = predict.run(images, properties, parameters)\n  File "/usr/local/lib/python3.10/dist-packages/vpt_plugin_cellpose/predict.py", line 34, in run\n    mask = model.eval(\n  File "/usr/local/lib/python3.10/dist-packages/cellpose/models.py", line 286, in eval\n    masks, flows, styles = self.cp.eval(x,\n  File "/usr/local/lib/python3.10/dist-packages/cellpose/models.py", line 629, in eval\n    masks, styles, dP, cellprob, p, bd, tr = self._run_cp(x,\n  File "/usr/local/lib/python3.10/dist-packages/cellpose/models.py", line 747, in _run_cp\n    p = np.array(p)\n'

2025-10-14 04:09:33,975 - ./task-260 - INFO - Run segmentation on tile 260 started
2025-10-14 04:09:33,988 - ./task-260 - INFO - Tile 260 [0, 59748, 5096, 5096]
2025-10-14 04:09:35,335 - ./task-288 - INFO - get_polygons_from_mask: z=5, labels:145
2025-10-14 04:09:38,017 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,017 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,019 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,019 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,020 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,021 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,021 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,021 - distributed.nanny - WARNING - Worker process still alive after 4.0 seconds, killing
2025-10-14 04:09:38,171 - . - ERROR - exception of type <class 'ValueError'> thrown: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (7, 2) + inhomogeneous part.
2025-10-14 04:09:38,171 - . - ERROR - vpt has encountered a runtime error (setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (7, 2) + inhomogeneous part.) and will now exit
/usr/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 8 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
`

The code I ran resulting in this error is as follows:
`#!/bin/sh
#BSUB -a 'docker(matinem/vpt-docker)'
#BSUB -B
#BSUB -G compute-hwgabel
#BSUB -J vpt.exp.37
#BSUB -N
#BSUB -n 8
#BSUB -oo /storage2/fs1/hwgabel/Active/Vizgen/cellpose.logs/vpt.exp.37.log
#BSUB -q general
#BSUB -R 'gpuhost'
#BSUB -gpu "num=2:gmodel=TeslaV100_SXM2_32GB"
#BSUB -R 'rusage[mem=128GB]'
#BSUB -u mnemera@wustl.edu

vpt --verbose --log-level 2 --processes 8 run-segmentation \
    --segmentation-algorithm /storage2/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/new_output/region_0/segmentation.json \
    --input-images /storage1/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/merfish_output/202412161325_exp37-Foxg1-resc-anterior_VMSC20502/region_0/images/ \
    --input-micron-to-mosaic /storage1/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/merfish_output/202412161325_exp37-Foxg1-resc-anterior_VMSC20502/region_0/images/micron_to_mosaic_pixel_transform.csv \
    --output-path /storage2/fs1/hwgabel/Active/Vizgen/experiment37_Foxg1_rescueAAV_anterior_2024-12-16/new_output/region_0/ \
    --tile-size 4096 \
    --tile-overlap 500 \
    --overwrite
`
Is there any advice on how to resolve this segmentation issue?
Best,
Mati

**Comment by MatiNem** [comment]
To anyone still dealing with this issue, Vizgen tech support helped me solve it by changing the tile size. Apparently your choice in tile size can lead to a bug in cellpose when reading in the mosaic images. I changed the tile size I was using from 4096 to 2048 and it worked.
